contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -495,24 +507,22 @@ contract FunctionsRouter is IFunctionsRouter, FunctionsSubscriptions, Pausable,
     for (uint256 i = 0; i < idsArrayLength; ++i) {
       bytes32 id = proposedContractSetIds[i];
       address proposedContract = proposedContractSetAddresses[i];
+
       if (
         proposedContract == address(0) || // The Proposed address must be a valid address
         s_route[id] == proposedContract // The Proposed address must point to a different address than what is currently set
       ) {
         revert InvalidProposal();
       }
-    }
 
-    s_proposedContractSet = ContractProposalSet({ids: proposedContractSetIds, to: proposedContractSetAddresses});
-
-    // NOTE: iterations of this loop will not exceed MAX_PROPOSAL_SET_LENGTH
-    for (uint256 i = 0; i < proposedContractSetIds.length; ++i) {
```

Comment:
what happened here?

Reply:
To save on contract size the second loop got combined in to the first one.
You'll see that the event gets emitted at the same time as validation, which is confusing, but if an error is hit then it will roll back all of the events.